### DISCUSSION TITLE
How do you select a cell with more than one inside of it?

### DISCUSSION THREAD

**Discussion by christianangon** [exceljs/exceljs#2004] [Q&A] [answered]
How do you properly select  a getCell with more than one cell inside of parenthesis?
My main objectives in here is to put some background color from cell A1 - F1
sample: A1 - F1
I tried this but I think most of it doesnt work.
 worksheet.getCell('A1:F1')
 worksheet.getCell('A1','F1')

**Comment by christianangon** [comment]
I tried instead using of getCell, I used getRows but apparently getRows(1, 6) is not working with the length.

**Comment by skypesky** [comment, marked as answer]
@spellman1980 
This may be useful to you.
```typescript
   const workbook = new Workbook(),
      worksheet = workbook.addWorksheet('test');

    // A1 -> F1
    new Array(6).fill(null).map((value, index) => {
      worksheet.getCell(1, index + 1).style.fill = {
        type: 'pattern',
        pattern: 'solid',
        fgColor: {
          argb: 'FFFF0000'
        }
      }
    });
  // export excel ...
```

**Reply by christianangon** [reply]
yeah thanks for your answer.
I resolve something but it is the same process
const cellBorder = ['A1', 'B1', 'C1', 'D1', 'E1', ''F1]
      cellBorder.forEach(col => {
        worksheet.getCell(col).fill= {
         type: 'pattern',
        pattern: 'solid',
        fgColor: {
          argb: 'FFFF0000'
        }
        }
      })

**Reply by christianangon** [reply]
Hi! Let's say I wanna select all cell in worksheet like ctrl+A command. How can I do that?

**Comment by skypesky** [comment]
@spellman1980 

What is the action after selecting all?

**Reply by christianangon** [reply]
To check all the cell value if its contain object, then return something inside of the object if it checks the if statement

**Comment by skypesky** [comment]
@spellman1980  
---
try it => `worksheet.eachCell()` ?
It  looks simple.if i am wrong.you can give me more specific test code.

**Reply by christianangon** [reply]
for (let i = 2; i < colsBorderCount; i++) {
    const rowsObjectCheck = worksheet.getRow(i)
    rowsObjectCheck.eachCell({ includeEmpty: false }, function(cell, rowNumber) {
      if (typeof cell._value.value === 'object') {
        if (cell._value.value.t) {
          cell._value.value = cell._value.value.t
          return
        } else {
          cell._value.value = ''
        }
      }
    })
  }

**Comment by skypesky** [comment]
@spellman1980 

It seems that your method is effective, right?
What are the Excel files you use to test and the expected program output?

```typescript
workbook.worksheets[0].eachRow((row: Row, rowNumber: number) => {
    const rowValues: CellValue[] = <CellValue[]>row.values;
    // coding ...
}
```

**Reply by christianangon** [reply]
@skypesky 
Well, my only problem in here is the date output. 
when I successfully return the item object, upon exporting, there's a double quotes at the beginning of the date ("2022-05-04") on my date column and writing it directly at JavaScript didn't work for it like using replace method or toString.
So what I did is instead I control the date format of it
1. I convert the date into new Date()
2. Convert it into into datenum function
3. Then use datenum for converting into date format

Code for datenum
```
function datenum(v) {
  var returnDateTime =
    25569.0 +
    (v.getTime() - v.getTimezoneOffset() * 60 * 1000) / (1000 * 60 * 60 * 24);
  return returnDateTime;
}
```
[Book1.xlsx](https://github.com/exceljs/exceljs/files/8617976/Book1.xlsx)

Then use this code for getting the date 
```
const changeTime = new Date(cell._value.value.t)
cell._value.value = datenum(changeTime)
```

and add this for date format
`[worksheet.getCell(cell._column._address).numFmt = 'yyyy-mm-dd']

And when I export the file it didnt convert the date. The output is not take effect unless you double click the cell first to convert into
'yyyy-mm-dd' date format. 

In the future I want to control my date format base on cell value length like
```
if(cell._value.value.t.length === 7){
worksheet.getColumn(cell._column._address).numFmt = 'yyyy-MM'
} else {
worksheet.getColumn(cell._column._address).numFmt = 'yyyy-MM-dd HH:mm:ss'
}
```

**Comment by skypesky** [comment]
@spellman1980 
---
I just tested it. It may be a bug I suggest using [dayjs](https://www.npmjs.com/package/dayjs) to format the date

@see more: https://github.com/exceljs/exceljs/issues/538
